sdcm/nemesis.py

```diff
@@ -1907,6 +1908,7 @@ def disrupt_grow_shrink_cluster(self):
                 self.metrics_srv.event_stop('del_node')
 
         self.log.info("Finish cluster shrink. Current number of nodes %s", len(self.cluster.nodes))
+        base_target_node.running_nemesis = None
```

Comment:
Since `self.target_node` is not used in this nemesis you can set `self.target_node.running_nemesis = None` immediately when the nemesis starts, this will prevent the node being falsely used by running nemesis.

**Alternatively**, and I think it is a better approach: instead of calling `self.set_target_node()` in the Shrink cluster step use the current target node, so clearing `running_nemesis` will be guaranteed by the Nemesis framework (look in the run method)

Reply:
@bentsi 
Alternative path will not work if nemesis will add >= 2 node. in this case nemesis should choose new node for next decommissioning and terminating.
From other hand, if disable running_nemesis on the beginning, then the logic of run will be changed, and the same node where GrowShrink nemesis started will used again in parallel nemesis. This could confuse when read logs. You will see that 2 nemesis will start on same node

Reply:
you are right regarding the alternative path. Let's go the first path, I agree that this can be misleading in the logs but we don't have solution for this in the current architecture, there are other nemesis that doesn't work on target node and some on all nodes, we can live with that.
Setting `self.target_node.running_nemesis = None` at the beginning of the nemesis will also prevent "locking" the node when exception happens in `_decommission_cluster_node`

Reply:
@bentsi thanks.

Reply:
Fixed